src/types/modules/module.cpp

```diff
@@ -403,4 +496,81 @@ void Module::_importJson(const Json& json)
 	}
 }
 
+/**
+ * A mapping converting a given string to corresponding Symbol.
+ */
+std::shared_ptr<Symbol> Module::_stringToSymbol (const std::shared_ptr<Symbol>& base, const std::string& str)
+{
+	std::string current_attribute = str;
+	std::string next_attribute;
+	auto delim_index = str.find('.');
+
+	if (delim_index != std::string::npos)
+	{
+		current_attribute = str.substr(0, delim_index);
+		next_attribute = str.substr(delim_index + 1, str.length());
+	}
+
+	if (!base)
+	{
+		if (_structure->getName() != current_attribute)
+			throw ModuleError("Unaccessible namespace. You can reference objects only from the same module." + getPathsAsString());
+
+		if (!next_attribute.empty())
+			return _stringToSymbol(_structure, next_attribute);
+		else
+			return _structure;
+	}
+	else
+	{
+		if (base->getType() != Symbol::Type::Structure)
+			throw ModuleError("Can't access an attribute of " + base->getName() + " as it's not a structured object." + getPathsAsString());
+
+		auto baseStruct = std::static_pointer_cast<StructureSymbol>(base);
+		auto found_symbol = baseStruct->getAttribute(current_attribute);
+
+		if (!found_symbol)
+			throw ModuleError("Object " + base->getName() + " does not contain an attribute with the name of " + current_attribute + "." + getPathsAsString());
+
+		auto current_symbol = found_symbol.value();
+
+		if (current_symbol->getType() == Symbol::Type::Dictionary ||
+			current_symbol->getType() == Symbol::Type::Array)
+		{
+			auto current_iterable = std::static_pointer_cast<IterableSymbol>(current_symbol);
+			current_symbol = current_iterable->getStructuredElementType();
+		}
+
+		if (!next_attribute.empty())
+		{
+			auto newBase = std::static_pointer_cast<StructureSymbol>(current_symbol);
+			return _stringToSymbol(newBase, next_attribute);
+		}
+
+		return current_symbol;
+	}
+}
+
+/**
+ * Adds an attribute to it's base that can be either a parent structure or an iterable.
+ */
+void Module::_addObjectToBase(Symbol* base, std::shared_ptr<Symbol> newAttribute)
+{
+	if (base)
+	{
+		switch (base->getType())
+		{
+			case Symbol::Type::Structure:
+				((StructureSymbol*)base)->addAttribute(newAttribute);
```

Comment:
Again quite a risky cast. In Yaramod we use `static_pointer_cast` to cast from pointer to derived class to pointer to base class which is safe, but not appliable here. If I am not mistaken, so far we did not need to cast base class to derived class. Is there any way we would avoid it here?

Reply:
As explained above, the cast is needed, because iterables and structures use different functions to get and set their attributes. I've changed the cast here to the same dynamic cast as in the other example, let me know if it's okay